### PR TITLE
fix(audit, auditTime): audit and auditTime emit last value after source completes

### PR DESCRIPTION
**Pull request by alexsherekin**
**Description:**
Adjust audit operator to emit the last value when source completes

**Related issue (if exists):**
#5730

### CODE REVIEW

File: spec/operators/audit-spec.ts
Comment (benlesh) [inline spec/operators/audit-spec.ts:177]: Hmmm... I'll need to think about this. I think this makes the most sense. Given that there are items to audit, and we're waiting for a duration that just never arrives... But I want a second opinion: @cartant

File: src/internal/operators/audit.ts
Comment (benlesh) [inline src/internal/operators/audit.ts:61]: @cartant, do we want to fix the duration notifier treating completion like a notification here? Or shall we get that at a later time? The whole thing is weird. Like `[]` "Here's your value, in this array!". My gut tells me it should be a `noop`, and I think we all agreed on that.

**Review by benlesh — CHANGES_REQUESTED** [review]
Just a few things, and some open questions for other core team members.

File: spec/operators/audit-spec.ts
Comment (cartant) [reply spec/operators/audit-spec.ts:177]: Yeah, I think this makes sense, too. Above all, the behaviour needs to be consistent between operators and, AFAICT, `delayWhen` doesn't complete if its notifier is `NEVER`: https://github.com/ReactiveX/rxjs/blob/0d3094f29b5f6f2cc4d2829f8a0e20addd2106a3/spec/operators/delayWhen-spec.ts#L131-L144

So this LGTM.

File: src/internal/operators/audit.ts
Comment (cartant) [reply src/internal/operators/audit.ts:61]: Yeah, we should fix it. Whether it's fixed here or someplace else, I don't mind. I intend to do another pass through all of the operators that take a notifier to make sure their behaviour is consistent, etc.

**Review by cartant — APPROVED** [review]
LGTM. @benlesh, I added responses to your comments.

**Review by benlesh — APPROVED** [review]

**Comment by benlesh** [comment]
Thanks, @alexsherekin!

**Comment by elepner** [comment]
Hello! This change broke our code after update. We use it to detect long press gesture and relied on the fact that audit does not emit last value if observable completes during audit window. Is it possible to add the same configuration as ``throttle`` or ``takeWhile`` operators do?

**Pull request merged**
